src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BlockSizeRule.cs

```diff
@@ -22,21 +22,25 @@ public override Task RunAsync(RuleContext context)
                 return Task.CompletedTask;
 
             var consensus = this.Parent.Network.Consensus;
+            
+            // An unexpected consequence of serialization is to change the Block.BlockSize property
+            // This rule serializes the block in a few ways to determine the block weight and as
+            // a result changes the BlockSize property unintentionally, to fix this we create a new instance of the Block
+            Block block = consensus.ConsensusFactory.CreateBlock();
```

Comment:
I think that creating a new block and serializing it is too heavy, aren't there any other ways to achieve this? 

Also it looks like it's a bug only when weight units are used (on segwit enabled networks), so this change would mean unnecessary performance hit for stratis

Reply:
Segwit will eventually come to stratis so it's relevant.

The other fix that was suggested to take the blocksize only once per serialisation.

I'll make new PR to demonstrate that.